### PR TITLE
Use webpack for browser builds

### PR DESCRIPTION
**Pull request by jdlehman**

### CODE REVIEW

**Comment by jdlehman** [comment]
@goatslacker here is a quick webpack config I through together, it probably still needs to be optimized some. Strangely it did not seem to help size at all.  I know like Babel, webpack also adds some helpers it uses so maybe that is adding extra weight?  I am no webpack expert so I could also be missing some configs.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  module: {
+    loaders: [
+      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ }
+    ]
+  },
+  output: {
+    path: __dirname + '/dist',
+    filename: getOutputFilename(production),
+    libraryTarget: 'umd',
+    library: 'alt'
```
Comment (goatslacker) [inline webpack.config.js]: does this mean it'll be available on `window.alt`?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  externals: [
+    {
+      "react": {
```
Comment (goatslacker) [inline webpack.config.js]: it's not excluding react

**Comment by goatslacker** [comment]
this is a good starting point

File: webpack.config.js (hunk 2 above)
Comment (troutowicz) [inline webpack.config.js]: I've always had a difficult time following webpacks documentation... but this works for me:

```js
externals: [ /^react(\/.*)?$/ ]
```

File: webpack.config.js (hunk 1 above)
Comment (jdlehman) [inline webpack.config.js]: This exports the whole bundle as a library so a user could just do `import 'alt';`

http://webpack.github.io/docs/configuration.html#output-library

File: webpack.config.js (hunk 2 above)
Comment (jdlehman) [inline webpack.config.js]: Oh probably because it isn't excluding the addons. Thanks @troutowicz, I'll fix that.

**Comment by david0178418** [comment]
This will be great as it seems that this won't require React to be global as it does now.

**Pull request closed without merging**